### PR TITLE
Dedupe points at query time if there are overlapping blocks

### PR DESCRIPTION
**Pull request by jwilder**

### CODE REVIEW

**Comment by jwilder** [comment]
@pauldix @otoolep

**Comment by pauldix** [comment]
I think it looks good, but any idea if there is any change in query performance in the non-duplicate case?

File: tsdb/engine/tsm1/file_store.go [hunk 1]
```diff
@@ -462,8 +479,20 @@ func (c *KeyCursor) init(t time.Time, ascending bool) {
...
+	c.ascending = ascending
 	c.seeks = c.fs.locations(c.key, t, ascending)
+
+	if len(c.seeks) > 0 {
+		for i := 1; i < len(c.seeks); i++ {
+			prev := c.seeks[i-1]
+			cur := c.seeks[i]
+
+			if prev.entry.MaxTime.Equal(cur.entry.MinTime) || prev.entry.MaxTime.After(cur.entry.MinTime) {
+				c.duplicates = true
```
Comment (otoolep) [inline tsdb/engine/tsm1/file_store.go]: Can this go false -> true -> false? You want it latched, right?

Comment (otoolep) [inline tsdb/engine/tsm1/file_store.go]: Scrap that, I see the break. :-)

**Comment by jwilder** [comment]
@pauldix Appears to be the same as before.

**Comment by otoolep** [comment]
Generally makes sense. I looked closely at the tests and the results seem right. +1

**Comment by otoolep** [comment]
Please confirm when cherry-picked to the 0.9.6 branch.

**Comment by jwilder** [comment]
Cherry-picked to 0.9.6

**Pull request merged**